pkg/tests/tasks/security/authorization/jwt_test.go

```diff
@@ -18,157 +18,196 @@ import (
 
-		groupURL := "https://raw.githubusercontent.com/istio/istio/release-1.9/security/tools/jwt/samples/groups-scope.jwt"
-		tokenGroup, err := util.ShellMuteOutput(`curl %s -s`, groupURL)
-		tokenGroup = strings.Trim(tokenGroup, "\n")
-		// tokenGroup, err = util.Shell(`echo %s | cut -d '.' -f2 - | base64 --decode -`, tokenGroup)
-		util.Inspect(err, "Failed to get JWT token", "", t)
-
-		log.Log.Info("Verify request with a JWT includes group1 claim")
-		cmd := fmt.Sprintf(`curl "http://httpbin.%s:8000/headers" -s -o /dev/null -H "Authorization: Bearer %s" -w "%%{http_code}\n"`, "foo", tokenGroup)
-		msg, err := util.PodExec("foo", sleepPod, "sleep", cmd, true)
-		util.Inspect(err, "Failed to get response", "", t)
-		if !strings.Contains(msg, "200") {
-			log.Log.Errorf("Verify request with JWT group1 claim Unexpected response: %s", msg)
-			t.Errorf("Verify request with JWT group1 claim Unexpected response: %s", msg)
-		} else {
-			log.Log.Infof("Success. Get expected response: %s", msg)
-		}
+		})
 
-		log.Log.Info("Verify request without groups claim JWT is denied")
-		cmd = fmt.Sprintf(`curl "http://httpbin.%s:8000/headers" -s -o /dev/null -H "Authorization: Bearer %s" -w "%%{http_code}\n"`, "foo", token)
-		msg, err = util.PodExec("foo", sleepPod, "sleep", cmd, true)
-		util.Inspect(err, "Failed to get response", "", t)
-		if !strings.Contains(msg, "403") {
-			log.Log.Errorf("Verify request without groups claim JWT Unexpected response: %s", msg)
-			t.Errorf("Verify request without groups claim JWT Unexpected response: %s", msg)
-		} else {
-			log.Log.Infof("Success. Get expected response: %s", msg)
-		}
+		t.NewSubTest("Security authorization allow JWT requestPrincipal").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, ns, JWTRequireRule)
+			})
+			oc.ApplyString(t, ns, JWTRequireRule)
+
+			t.LogStep("Verify that a request with a valid JWT is allowed")
+			assertRequestJWTAccepted(t, ns, curl_JWT("Authorization: Bearer %s", token))
+
+			t.LogStep("Verify request without a JWT is denied")
+			assertRequestJWTDenied(t, ns, curl_JWT())
+
+		})
+
+		t.NewSubTest("Security authorization allow JWT claims group").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, ns, JWTGroupClaimRule)
+			})
+			oc.ApplyString(t, ns, JWTGroupClaimRule)
+
+			t.LogStep("Verify that a request with the JWT that includes group1 in the groups claim is allowed")
+			assertRequestJWTAccepted(t, ns, curl_JWT("Authorization: Bearer %s", tokenGroup))
+
+			t.LogStep("Verify that a request with a JWT, which does not have the groups claim is rejected")
+			assertRequestJWTDenied(t, ns, curl_JWT("Authorization: Bearer %s", token))
+		})
+	})
+}
+
+func curl_JWT(headers ...string) string {
+	headerArgs := ""
+	for _, header := range headers {
+		headerArgs += fmt.Sprintf(` -H "%s"`, header)
+	}
+	return fmt.Sprintf(`curl "http://httpbin:8000/headers" -sS -o /dev/null %s -w "%%%%{http_code}\n"`, headerArgs)
+}
```

Comment:
This duplicates the code in httpbinRequest(). Instead of duplicating the code, you should call that function.

```suggestion
    httpbinRequest("GET", "/headers", headers...)
```

But I think you should just delete the `curl_JWT` function and call `httpbinRequest()` directly.